lib/src/builder/build_step.dart

```diff
@@ -13,6 +13,12 @@ abstract class BuildStep {
   /// The primary input for this build step.
   Asset get input;
 
+  /// Checks if an [Asset] by [id] exists as an input for this [BuildStep].
+  ///
+  /// If [trackAsDependency] is true, then [id] will be marked as a dependency
+  /// of all [expectedOutputs].
+  Future<bool> hasInput(AssetId id, {bool trackAsDependency: true});
```

Comment:
Is `trackAsDependency` something we want to expose to users, or should it be an implementation detail?

I feel like you had a use case for it, but I can't remember what it is right now.